packages/flame/lib/src/components/position_component.dart

```diff
@@ -164,6 +164,31 @@ class PositionComponent extends Component {
   Vector2 get scaledSize =>
       Vector2(width * scale.x.abs(), height * scale.y.abs());
 
+  final _totalScale = Vector2.zero();
+
+  /// The resulting scale after all the ancestors and the components own scale
+  /// has been applied.
+  Vector2 get absoluteScale {
```

Comment:
I don't think this is well-defined.
If component N is rotated 90º and is inside another component that scales x-direction twice, then the "absolute scale" of N is that its y-direction is stretched twice. Of course, it's even more complicated when the rotation angles are not nice.

Reply:
That should be fine? I scale the vertices first and then rotate them as separate actions, and everything is defined around the center.

Reply:
Rotations tend to turn X axes into Y and vice versa. So, scales from multiple transforms do not compose with each other.

Reply:
Ah right, I see what you mean... Hmm, seems like we'll have to introduce a matrix that we can accumulate all operations on then, that should work right? This seems like it will be quite an expensive operation.

Reply:
Indeed, rather expensive.

For a short-term solution we could use `PositionComponent.toAbsoluteRect()` which computes smallest bounding rect (in absolute coords) that circumscribes the component.

Alternatively, we could compute the product of `max(abs(scale.x), abs(scale.y))`, which will produce an over-estimation of the size of the component, but may be ok if we only need this in order to filter out objects that are sufficiently far away from each other.

Reply:
Since the collision detection system isn't only for FCS `PositionComponent.toAbsoluteRect()` can't be used, and it wouldn't be accurate enough.

I think that as a short-term solution we'll just say in the docs that we don't support ancestors of the parent to the hitbox that are scaled, so the direct parent can be scaled, but none above that level. I think this will be more useful since nobody have actually complained about that usecase so far.